### DISCUSSION TITLE
Automatic type conversion for get-function

### DISCUSSION THREAD

**Discussion by JHerrKodo** [nlohmann/json#3334] [Q&A] [answered]
Hello,

when accessing a value, you have to know the type of it:

```
  json j1 = {
    {"pi", 3.141}
  };

  double v_floating = j1.value( "pi", 3.14 );
```

I want to know if it is possible to get a value automatic converted to the type of the variable.
In the example below you can see how an automatic conversion could look like:

```
   std::string v_str = j1.value( "pi", "3.14" );
```

This is throwing an exception. Is there a way to emulate this functionality?

**Comment by nlohmann** [comment, marked as answer]
No, there is no such automatic conversion from `double` to `std::string`. You would need to define your own function for this.

**Reply by JHerrKodo** [reply]
Thanks for your reply. Meanwhile I read the documentation in detail and I recognized that it is essential to check typ first before extracting the value. In my mind it is a disadvantage that your library is less robust that others. I prefer methods which do not throw exeption but return default values, empty values or automaticed casted values. But this is a matter of philosophy. 
Nevertheless I like your library very much.

**Reply by nlohmann** [reply]
I would not call it less robust - rather more explicit. What API would you propose instead?

**Reply by JHerrKodo** [reply]
Yes maybe robust is the wrong word. But the user has to take more attention to the JSON file and its data. This does not have to mean anything bad. I'm working with your API, because the advantages still outweigh :)
